### PR TITLE
KAFKA-6637: Avoid divide /zero error with segment.ms set to zero

### PR DESCRIPTION
**Pull request by huxihx**
https://issues.apache.org/jira/browse/KAFKA-6637

This patch enforces a minimum value of 1 for the user-custom `segment.ms`.

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by huxihx** [comment]
@ijuma please review this minor patch. Thanks.

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 1]
```diff
@@ -1088,7 +1088,10 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean, dynamicConfigO
   def logIndexSizeMaxBytes = getInt(KafkaConfig.LogIndexSizeMaxBytesProp)
   def logIndexIntervalBytes = getInt(KafkaConfig.LogIndexIntervalBytesProp)
   def logDeleteDelayMs = getLong(KafkaConfig.LogDeleteDelayMsProp)
-  def logRollTimeMillis: java.lang.Long = Option(getLong(KafkaConfig.LogRollTimeMillisProp)).getOrElse(60 * 60 * 1000L * getInt(KafkaConfig.LogRollTimeHoursProp))
+  def logRollTimeMillis: java.lang.Long = Option(getLong(KafkaConfig.LogRollTimeMillisProp)) match {
```
Comment (omkreddy) [inline core/src/main/scala/kafka/server/KafkaConfig.scala]: we have condition below to check "log.roll.ms must be equal or greater than 1"
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaConfig.scala#L1306

I think we can set atleast(1) validator to "segment.ms" config here: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/log/LogConfig.scala#L217

**Comment by huxihx** [comment]
@omkreddy Please review this minor change. Thanks.

**Comment by omkreddy** [comment]
LGTM.

**Comment by huxihx** [comment]
@omkreddy Do you think it's okay to have it passed?

**Review by hachikuji — APPROVED** [review]
LGTM. Thanks for the patch.

**Pull request merged**